src/main/java/com/linkedin/kmf/apps/MultiClusterMonitor.java

```diff
@@ -29,17 +30,18 @@
 public class MultiClusterMonitor implements App {
   private static final Logger LOG = LoggerFactory.getLogger(MultiClusterMonitor.class);
 
-  private final MultiClusterTopicManagementService _topicManagementService;
+  private final MultiClusterTopicManagementService _multiClusterTopicManagementService;
   private final ProduceService _produceService;
   private final ConsumeService _consumeService;
   private final String _name;
 
   public MultiClusterMonitor(Map<String, Object> props, String name) throws Exception {
     _name = name;
     MultiClusterMonitorConfig config = new MultiClusterMonitorConfig(props);
-    _topicManagementService = new MultiClusterTopicManagementService(createMultiClusterTopicManagementServiceProps(props, config), name);
+    _multiClusterTopicManagementService = new MultiClusterTopicManagementService(createMultiClusterTopicManagementServiceProps(props, config), name);
+    CompletableFuture<Void> topicPartitionReady = _multiClusterTopicManagementService.topicPartitionReady();
     _produceService = new ProduceService(createProduceServiceProps(props, config), name);
-    _consumeService = new ConsumeService(createConsumeServiceProps(props, config), name);
+    _consumeService = new ConsumeService(createConsumeServiceProps(props, config), name, topicPartitionReady);
```

Comment:
Why does `consumeService` accept the future but not `produceService`?

Reply:
The new AdminClient, which is eventually consistent, renders a race condition for consumeService, wherein the produce availability indicates 1 but the consumer availability shows 0 availability without eventually converging. Thus, the topic partitions are ready in time for produceService but not for consumeService.

Reply:
AdminClient topic creation is relatively slow and and thus is a cause for the race condition problems.